src/components/nearMe/NearMeProfiles.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react'
+import { Box, Grid, ImageList, ImageListItem, Typography } from '@mui/material'
+import { LocationOn } from '@mui/icons-material'
+import { makeStyles } from 'tss-react/mui'
+import { getUsersNearMeData } from '../../actions/userServices'
+import { getColumns } from './helper'
+
+type isMobileProps = {
+  isMobile: boolean
+}
+type userNearMeObjectType = {
+  id: string
+  location: string
+  picture: string
+  fName: string
+  lName: string
+}
+const NearMeProfiles = ({ isMobile }: isMobileProps) => {
+  const { classes } = useStyles()
+  const [list, setList] = useState<Array<userNearMeObjectType>>([])
+  const columns = getColumns(isMobile)
+  const txtAlign = isMobile ? 'center' : 'left'
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const result = await getUsersNearMeData()
+        setList(result)
+      } catch (error) {
+        console.error('Error fetching data:', error)
+      }
+    }
+    fetchData()
+  }, [])
+  return (
+    <Grid container spacing={3}>
+      <Grid item xs={12}>
+        <Typography className={classes.headerNear} sx={{ textAlign: txtAlign }}>
+          Near by
+        </Typography>
+        <Typography
+          className={classes.description}
+          sx={{ textAlign: txtAlign }}
+        >
+          These people near you – just like them and see if it’s a match!
+        </Typography>
+      </Grid>
+      <Grid item xs={12}>
+        <ImageList cols={columns}>
+          {list
+            .filter((user) => user.location === 'Istanbul')
```

Comment:
Filter and map should be moved out . Sorry, maybe I was not clear at the previous iteration

Reply:
Moved out the .map and .filter part of the component to an external component UserListRenderer.tsx.
Created userTypes.ts to store types for UserListRenderer and NearMeProfiles components.